elastic/kibana
# Remove TLS requirement for alerting when security is enabled

Resolves #111721

With https://github.com/elastic/elasticsearch/pull/76801 now merged, alerting now works without TLS being enabled. The remaining pieces for us to change are our documentation and UI messages whenever Kibana fails to create API keys.

Note that our logic is still necessary when `xpack.security.authc.api_key.enabled: false` is set. But this is less likely than TLS not being set up.

The PR does the following:
- Update documentation to remove set up requirement of TLS
- Update documentation to mention API keys cannot be disabled, otherwise alerting will not work
- Update UI messages to mention API keys must be enabled (instead of TLS)

Follow ups:
- https://github.com/elastic/kibana/issues/115236

Previews:
- Docs: https://kibana_115234.docs-preview.app.elstc.co/diff

<details>
<summary>API keys disabled message</summary>

<img width="1080" alt="Screen Shot 2021-10-19 at 10 56 27 AM" src="https://user-images.githubusercontent.com/3694571/137936572-2d5050e4-f97c-4086-914c-a5ab0b26663f.png">

</details>

<details>
<summary>API keys and encryption key message</summary>

<img width="1074" alt="Screen Shot 2021-10-19 at 10 50 05 AM" src="https://user-images.githubusercontent.com/3694571/137936634-81ee9c91-c737-4432-8d4e-84a7902a8788.png">

</details>

<details>
<summary>Encription key message</summary>

<img width="1072" alt="Screen Shot 2021-10-19 at 10 54 23 AM" src="https://user-images.githubusercontent.com/3694571/137936739-22f96765-82cc-4bae-8c0c-ed8b37770e1e.png">

</details>

- Note: The messages are the same in the create rule flyout. To test, you'll need to access the flyout from an app.

## Testing steps

Explicitly disabling API keys
1. `yarn es snapshot -E xpack.security.authc.api_key.enabled: false` and `yarn start`
2. Notice the new banners when navigating to the rules management page

Ephemeral encryption key + disabling API keys
1. Change https://github.com/elastic/kibana/blob/master/x-pack/plugins/encrypted_saved_objects/server/plugin.ts#L61 to `const canEncrypt = false;`
2. `yarn es snapshot -E xpack.security.authc.api_key.enabled=false` and `yarn start`
3. Notice the new banners when navigating to the rules management page
